wp-content/plugins/tapestry/endpoints.php

```diff
@@ -3,3 +3,73 @@
  * Tapestry Endpoints
  *
  */
+
+require __DIR__ . '/controller/class.user-controller.php';
+
+ // User endpoints
+
+// Get user's progress
+add_action( 'rest_api_init', function () {
+    register_rest_route( 'tapestry-tool/v1', 'users/progress/', array(
+      'methods' => 'GET',
+      'callback' => 'tapestry_get_user_progress_by_post_id',
+    ) );
+});
+
+// Get user progress on a tapestry page by user id and post id. Will need to pass these as query parameters
+// Example: http://localhost:8888/tapestry-wp/wp-json/tapestry-tool/v1/users/progress?user_id=1&post_id=42
+function tapestry_get_user_progress_by_post_id($data) {
+    $userId = $data['user_id'];
+    $postId = $data['post_id'];
+    $userController = new TapestryUserController;
+    return $userController->getProgress($userId, $postId);
+}
+
+add_action( 'rest_api_init', function () {
+    register_rest_route( 'tapestry-tool/v1', '/users/progress/', array(
+      'methods' => 'POST',
+      'callback' => 'tapestry_update_user_progress_by_post_id',
+    ) );
+});
+
+// Example: http://localhost:8888/tapestry-wp/wp-json/tapestry-tool/v1/users/progress?user_id=1&post_id=44&json={"32":0.5,"33":0,"34":0,"35":0,"36":0,"37":0,"38":0,"39":0,"40":0,"41":0}
+function tapestry_update_user_progress_by_post_id($data) {
+    $userController = new TapestryUserController;
+    $userId = $data['user_id'];
```

Comment:
I don't think we need to pass in the user_id here since we have access to it through `get_current_user()` function. I know you are validating that the user passed in is the same user as the current user, but I don't see why we need to pass it in in the first place.

Reply:
get_current_user() seemed to always return 0 for me. Using apply_filters('determine_current_user', false) instead.